packages/ember/tests/routing/decoupled_basic_test.js

```diff
@@ -326,4 +327,217 @@ class extends ApplicationTestCase {
       );
     });
   }
+
+  [`@test The route controller specified via controllerName is used in render`](assert) {
+    this.router.map(function() {
+      this.route('home', { path: '/' });
+    });
+
+    this.add('route:home', Route.extend({
+      controllerName: 'myController',
+      renderTemplate() {
+        this.render('alternative_home');
+      }
+    }));
+
+    this.add('controller:myController', Controller.extend({
+      myValue: 'foo'
+    }));
+
+    this.addTemplate('alternative_home', '<p>alternative home: {{myValue}}</p>');
+
+
+    return this.visit('/').then(() => {
+      let homeRoute = this.applicationInstance.lookup('route:home');
+      let myController = this.applicationInstance.lookup('controller:myController');
+      let text = this.$('p').text();
+
+      assert.deepEqual(homeRoute.controller, myController,
```

Comment:
This was `deepEqual` in the old test, but you almost certainly want `equal` to confirm they are the *same* instance. Please use `equal`.

I'm surprised this works with `myController` instead of `my-controller`. I expected the test mode resolver would fail with camelcase!